src/FilenameParsing/AbstractFileIDHelper.php

```diff
@@ -53,6 +73,66 @@ public function buildFileID($filename, $hash = null, $variant = null, $cleanfile
         return $fileID;
     }
 
+    /**
+     * Get the original filename with the extension rewritten to be the same as the variant file extension.
+     *
+     * @param string $filename Original filename without variant
+     */
+    public function rewriteVariantExtension(string $filename, string $variant): string
+    {
+        return $this->swapExtension($filename, $variant, self::EXT_VARIANT);
+    }
+
+    /**
+     * Get the variant filename with the extension rewritten to be the same as the original file extension.
+     *
+     * @param string $filename Original filename without variant
+     */
+    public function restoreOriginalExtension(string $filename, string $variant): string
+    {
+        return $this->swapExtension($filename, $variant, self::EXT_ORIGINAL);
+    }
+
+    /**
+     * Get the original file's filename with the extension rewritten to be the same as either the original
+     * or the variant extension.
+     *
+     * @param string $filename Original filename without variant
+     * @param int $extIndex One of self::EXT_ORIGINAL or self::EXT_VARIANT
+     */
+    private function swapExtension(string $filename, string $variant, int $extIndex): string
+    {
+        // If there's no variant at all, we can rewrite the filenmane
+        if (empty($variant)) {
+            return $filename;
+        }
+
+        // Split variant string in variant list
+        $subVariants = explode('_', $variant);
```

Comment:
What happens when the original filename has underscores in it? You'll need to add a unit test for this.

Reply:
It won't match the variant regex and so it will be ignored.

Reply:
I've added underscores to file names in the test scenarios.